R/forecast.R

```diff
@@ -65,6 +65,9 @@ as_forecast <- function(data,
 #' @param sample_id (optional) Name of the column in `data` that contains the
 #'   sample id. This column will be renamed to "sample_id". Only applicable to
 #'   sample-based forecasts.
+#' @param check_forecast_type Logical, defaults to `FALSE`. Should the
+#' `forecast_type` be checked and reset if not as implied by the `data`?
```

Comment:
At the moment this throws an error if not as implied by the data. I think the re-setting is only done if `forecast_type = NULL` as only then the `assert_` call succeeds?

```r
> as_forecast(example_point, check_forecast_type = TRUE, forecast_type = "quantile")
Error in `assert_forecast_type()` at scoringutils/R/forecast.R:127:5:
! Forecast type determined by scoringutils based on input: "point".
i Desired forecast type: "quantile".
Run `rlang::last_trace()` to see where the error occurred.
```

Reply:
I'm not entirely following what the issue is that prompted this change and why we need the `check_forecast_type` argument. 
Is the `assert_forecast_type(data, desired = forecast_type)` check annoying? Because that should always succeed if the user specified `forecast_type = NULL` (i.e. the default). 
Or is the check with the old format annoying? My intuition is that that check should happen by default.

Reply:
> I'm not entirely following what the issue is that prompted this change and why we need the check_forecast_type argument.

I was trying to use `as_forecast` to reclass an object and all these checks get in the way of that (and in general get in the way of generalising to new classes without monolithic changes). I had to drop doing that as even with above it wouldn't work. More generally I think we need some kind of skipping option for people writing their own classes who want to still use `as_forecast` around new_forecast as these checks may not work.

Reply:
Rewordeed docs